drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -286,6 +346,14 @@ static int nrf5_energy_scan_start(const struct device *dev,
 		if (nrf_802154_energy_detection(duration * 1000) == false) {
 			nrf5_data.energy_scan_done = NULL;
 			err = -EPERM;
+		} else {
```

Comment:
There are several places in this PR where this pattern:
```c
    } else {
#if (CONFIG_IEEE802154_NRF5_CALLBACK_TIMEOUT_MS != 0)
        // foo
#endif
    }
```
is used. It's tempting to replace it with:
```c
    } else if (CONFIG_IEEE802154_NRF5_CALLBACK_TIMEOUT_MS) {
        // foo
    }
```
It looks cleaner and might make maintenance easier because compilation errors will not slip through even if the feature is disabled by default. Please check if such a change is feasible (might not be if the code refers to variables that must be `#ifdef'ed`).

Reply:
I think I fixed as you pointed, please have a look

Code related to `ed_timer` must remain, the `ed_timer` is in the struct. If I left `ed_timer` unconditionally, for sake of having all paths compiled, there would be a waste of ram in case ed_timer is not used.

Reply:
That makes sense, thanks.